### PR TITLE
Benchmark chart components

### PR DESCRIPTION
**Pull request by aaron-russell**
#### Tests
Does this PR include tests for the changes introduced?
- [ ] Yes
- [X] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { BarchartComponent } from './barchart.component';
+
+fdescribe('BenchmarksTabComponent', () => {
```
Comment (madeviv) [inline src/app/shared/components/benchmark-metric/barchart/barchart.component.spec.ts:5]: don't forget to remove

File: src/app/shared/components/benchmarks-tab/benchmarks.module.ts [hunk 2]
```diff
@@ -6,16 +6,10 @@ import { BenchmarksRoutingModule } from '@shared/components/benchmarks-tab/bench
...
-  imports: [CommonModule, RouterModule, OverlayModule,BenchmarksRoutingModule],
-  declarations: [
-    BenchmarksAboutTheDataComponent,
-  ],
-  exports: [
-    BenchmarksAboutTheDataComponent,
-  ],
-  providers: [
-  ],
+  imports: [CommonModule, RouterModule, OverlayModule, BenchmarksRoutingModule],
```
Comment (madeviv) [inline src/app/shared/components/benchmarks-tab/benchmarks.module.ts:10]: Does this work without CommonModule and RouterModule?

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  'pay',
+  'turnover',
+  'qualification',
+  'sickness',
+}
+
+@Directive({
+  selector: 'metric-desc',
+})
+export class MetricDescDirective {}
```
Comment (madeviv) [inline src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:14]: What does this do?

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+          accessibility: {
+            description: this.altdescription,
+            enabled: true,
+          },
+          data: [
+            { y: this.yourworkplace, color: '#28a197', name: 'Your workplace' },
+            { y: this.comparisongroup, name: 'Comparison group' },
+            { y: this.goodcqc, name: 'Good and outstanding CQC providers in your comparison group' },
+            { y: this.lowturnover, name: 'Workplaces with a low turnover rate in your comparison group' },
+          ],
```
[the remark this answers is not in the document]
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:101]: Not really no, that's the unfortunate thing about highcharts

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                  break;
+                case Metric.sickness:
+                  value = this.y + ' days';
+                  break;
+                default:
+                  value = this.y + '%';
+              }
+              const size = this.key === 'Your workplace' ? 'govuk-heading-xl' : 'govuk-heading-m';
+              return '<span class="' + size + ' govuk-!-margin-bottom-2">' + value + '</span>';
+            },
```
[the remark this answers is not in the document]
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:123]: Not quite sure what you mean?

File: src/app/shared/components/benchmark-metric/benchmark-metric.module.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { OverlayModule } from '@angular/cdk/overlay';
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { RouterModule } from '@angular/router';
```
[the remark this answers is not in the document]
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/benchmark-metric.module.ts:4]: Good spot

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
+import { Component, Directive, Input, OnInit } from '@angular/core';
+import * as Highcharts from 'highcharts';
+
+export enum Metric {
+  'pay',
+  'turnover',
+  'qualification',
+  'sickness',
+}
```
[the remark this answers is not in the document]
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:9]: I wasn't quite sure to put this, It's probably better with the models. I'll put it there, thanks!

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts (hunk 3 above)
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:14]: That's the the directive, so you can have a `<metric-desc>` element, and that can change depending on some ifs in the view. It's similar to how the tiles are built. It gives you a little more flexibility on that to show.

File: src/app/shared/components/benchmark-metric/barchart/barchart.component.ts [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+                  .add();
+                text.attr({
+                  x: this.plotLeft + offset - text.getBBox().width / 2,
+                  y: this.plotTop + (this.plotHeight / 3) * 2,
+                });
+              }
+            });
+          },
+        },
+      },
```
[the remark this answers is not in the document]
Comment (aaron-russell) [reply src/app/shared/components/benchmark-metric/barchart/barchart.component.ts:89]: Which logic are you referring to sorry? As it's showing lines 86 - 89 which is just closing braces?

**Review by madeviv — APPROVED** [review]

**Pull request merged**